dashboard/api.py

```diff
@@ -32,6 +33,10 @@
 COURSE_GRADE_WEIGHT = 0.4
 EXAM_GRADE_WEIGHT = 0.6
 
+CACHE_KEY_FAILURE_NUMS_BY_USER = "update_cache_401_failure_numbers"
```

Comment:
What's the difference between these two keys? It seems like if we store the number of failures we can put the logic to determine if that's too many in the place where we do the update. We don't need an extra field showing whether or not to update the user

Reply:
Can you add a comment for each of these keys describing what information is stored there?